packages/zendesk-adapter/src/change_validators/one_translation_per_locale.ts

```diff
@@ -90,18 +102,19 @@ export const oneTranslationPerLocaleValidator: ChangeValidator = async (changes,
         CATEGORY_TRANSLATION_TYPE_NAME,
         SECTION_TRANSLATION_TYPE_NAME].includes(instance.elemID.typeName))
 
-  // it is a set to avoid duplicate of parents from translations which have the same parent
-  const parentInstances = relevantInstances
+  const parentInstancesId = relevantInstances
     .filter(instance => isTranslation(instance))
-    .flatMap(getParents)
-    .map(translationParentInstance => translationParentInstance.value)
+    .flatMap(instance => instance.annotations[CORE_ANNOTATIONS.PARENT])
+    .flatMap(parent => (isReferenceExpression(parent) ? parent.elemID : undefined))
+    .filter(isDefined)
 
-  const parentInstancesObj = _.keyBy(parentInstances, instance => instance.elemID.getFullName())
+  const parentInstances = await awu(parentInstancesId)
+    .map(async id => elementSource.get(id))
+    .filter(isInstanceElement)
+    .toArray()
 
-  if (elementSource === undefined) {
-    log.error('Failed to run findDuplicateTranslations because no element source was provided')
-    return []
-  }
+  // it is a record to avoid duplicate of parents from translations which have the same parent
+  const parentInstancesObj = _.keyBy(parentInstances, instance => instance.elemID.getFullName())
```

Comment:
if the goal is uniqueness, I think we will benefit more from dedup'ing before calling the element source - maybe use
```suggestion
  const parentInstances = await awu(parentInstancesId)
    .uniqBy(id => id.getFullName())
    .map(async id => elementSource.get(id))
    .filter(isInstanceElement)
    .toArray()
```
and then we won't need the `parentInstancesObj` at all?

Reply:
sounds good! it didn't work exactly like that so I did something a bit different.